oidc-exchange.py

```diff
@@ -0,0 +1,65 @@
+import os
+import sys
+
+import requests
+
+
+def die(msg: str):
+    print(f"OIDC exchange failure: {msg}", file=sys.stderr)
+    sys.exit(1)
+
+
+oidc_audience = os.getenv("INPUT_OIDC_AUDIENCE")
+
+oidc_token_url = os.getenv("ACTIONS_ID_TOKEN_REQUEST_URL")
+if oidc_token_url is None:
+    die("no ACTIONS_ID_TOKEN_REQUEST_URL in environment")
+
+oidc_bearer_token = os.getenv("ACTIONS_ID_TOKEN_REQUEST_TOKEN")
+if oidc_bearer_token is None:
+    die("no ACTIONS_ID_TOKEN_REQUEST_TOKEN in environment")
+
+oidc_resp = requests.get(
+    oidc_token_url,
+    params={"audience": oidc_audience},
+    headers={"Authorization": f"bearer {oidc_bearer_token}"},
+)
+
+if not oidc_resp.ok:
+    die(f"OIDC credential request error: {oidc_resp.status_code}")
```

Comment:
This will probably be a common failure for anyone who forgot to set `id-token: write`, we could probably be more specific here in that instance.

(We should also make sure to document that requirement in the docs for this action)

Reply:
(On second though, I'm not sure if it will fail here or above when we check for `ACTIONS_ID_TOKEN_REQUEST_URL` or `ACTIONS_ID_TOKEN_REQUEST_TOKEN`.)